interp/interp.go

```diff
@@ -452,8 +452,8 @@ func (interp *Interpreter) EvalWithContext(ctx context.Context, src string) (ref
 		interp.stop()
 		return reflect.Value{}, ctx.Err()
```

Comment:
How about
```suggestion
		v, err = reflect.Value{}, ctx.Err()
```
instead? That would make L456 the unique return point, which would be more readable and elegant imho.

Reply:
This change causes a race condition, due to the concurrency with line 447.